django/conf/__init__.py

```diff
@@ -105,6 +105,21 @@ def __init__(self, settings_module):
                 setattr(self, setting, setting_value)
                 self._explicit_settings.add(setting)
 
+        if self.SECRET_KEY_FILE and not self.SECRET_KEY:
+            import stat as s
+
+            # Verify that file permissions are set to be
+            # only user readable and writable.
+            mode = os.stat(self.SECRET_KEY_FILE).st_mode
+            if bool((s.S_IRGRP | s.S_IWGRP | s.S_IXGRP | s.S_IROTH | s.S_IWOTH | s.S_IXOTH) & mode):
```

Comment:
@berkerpeksag This code is never used outside of `django.conf` so I feel that moving it into `django/utils/secret_key.py` will just add unnecessary complexity